src/tools/crossgen2/Common/TypeSystem/Common/Utilities/TypeNameFormatter.cs

```diff
@@ -105,7 +127,7 @@ public TState AppendName(StringBuilder sb, TypeDesc type, TOptions options)
                 case TypeFlags.SignatureMethodVariable:
                     return AppendName(sb, (SignatureMethodVariable)type, options);
                 default:
-                    Debug.Assert(type.IsDefType);
+                    Debug.Assert(normalCategoryComputation && type.IsDefType); // Don't call type.IsDefType if Category computation failed
```

Comment:
Let's just delete the assert. The cast below is going to fail anyway.

Reply:
True, its really testing does IsDefType work, and that's well known to be true.